src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -2266,13 +2268,21 @@ module Base = struct
                Amount.(var_of_t zero))) ;
       with_label __LOC__ (fun () ->
           run_checked
-            (Fee_excess.assert_equal_checked statement.fee_excess
-               { fee_token_l = Token_id.(var_of_t default)
-               ; fee_excess_l =
-                   Amount.Signed.Checked.to_fee (fst init).fee_excess
-               ; fee_token_r = Token_id.(var_of_t default)
-               ; fee_excess_r = Amount.Signed.Checked.to_fee global.fee_excess
-               })) ;
+            (let rebalanced fee_excess =
+               let%bind f = Fee_excess.to_field_var fee_excess in
+               Fee_excess.rebalance_checked f
+             in
+             let%bind expected = rebalanced statement.fee_excess in
+             let%bind got =
+               rebalanced
+                 { Fee_excess.fee_token_l = Token_id.(var_of_t default)
+                 ; fee_excess_l =
+                     Amount.Signed.Checked.to_fee (fst init).fee_excess
```

Comment:
When you say part-way through a block, do you mean when we split a parties transaction across multiple scan state leaves? 
`(fst init).fee_excess` will be zero because of this https://github.com/MinaProtocol/mina/blob/95b5db0b0c54b133163a6e9c8711d9259d8096a0/src/lib/transaction_snark/transaction_snark.ml#L3908
I'll remove the rebalancing

Reply:
No, I mean that it isn't the first transaction in the block. If it were e.g. the second, then the `(fst init).fee_excess` will be the non-zero excess from the previous transaction, and the code as written will effectively double it.

Reply:
base statements don't carry previous transactions' fee excess.
https://github.com/MinaProtocol/mina/blob/af76cb7980d5e81e704120290a850ea9c6f8522e/src/lib/staged_ledger/staged_ledger.ml#L479

Reply:
By my understanding, parties logic behaves differently, and this is broken by that interaction. To be absolutist: any statement of any non-merge that has more than a single excess is an error.

Reply:
After our chat, I'm happy that the difference calculation below correctly accounts for this.